.github/workflows/build.yaml

```diff
@@ -21,9 +21,13 @@ jobs:
         run: |
           go fmt ./...
           go vet ./...
+          test -z "$(git status --porcelain)" || (echo "please run 'go fmt ./... && go vet ./...', and submit your changes"; exit 1)
```

Comment:
I prefer `git diff` here. It will show you exactly what lines you missed - which I think is useful information to have when troubleshooting why `go fmt` is not happy in CI (while it may work on my machine).

In contrast to `go mod vendor`, I don't think `go fmt` or` go vet` can create any files, so I don't think `git diff` has any downsides over `git status` here.

Reply:
Make complete sense, will update the PR accordingly!